scylla/src/utils/mod.rs

```diff
@@ -2,3 +2,11 @@ pub(crate) mod parse;
 
 pub(crate) mod pretty;
 pub mod test_utils;
+
+// FIXME: replace this with `Option::unzip()` once MSRV is bumped to at least 1.66
+pub(crate) fn unzip_option<T, U>(opt: Option<(T, U)>) -> (Option<T>, Option<U>) {
+    match opt {
+        Some((a, b)) => (Some(a), Some(b)),
+        None => (None, None),
+    }
+}
```

Comment:
Instead of a standalone function, you can implement an extension trait for `Option` which defines an `unzip` function for it. The code will look like you used a real `Option::unzip` in the standard library. When we bump the MSRV you will only have to drop the trait import.

Reply:
The idea is great, but not achievable...
```
error[E0658]: use of unstable library feature 'unzip_option': recently added
   --> scylla/src/transport/iterator.rs:249:14
    |
249 |             .unzip();
    |              ^^^^^
    |
    = note: see issue #87800 <https://github.com/rust-lang/rust/issues/87800> for more information
```
even though the trait is imported by the module. For this to work, fully qualified syntax is needed:
```rust
<Option<(PartitionKey, Token)> as UnzipOption<PartitionKey, Token>>::unzip(
            prepared
                .extract_partition_key_and_calculate_token(&serialized_values)?);
```

So, I'm keeping the standalone function.

Reply:
Bummer.